app/controllers/relationships_controller.rb

```diff
@@ -0,0 +1,29 @@
+class RelationshipsController < ApplicationController
+  before_action :logged_in_user
+
+  def index
+    @user = User.find_by id: params[:id]
+    redirect_to root_path unless @user
```

Comment:
You should create method like this:
```
private

def load_user
  @user = User.find_by id: params[:id] || params[:followed_id]
  return if @user.present?

  flash...
  redirect_to root_path
end
```
Then create before_action for 2 action 'index' and 'create'